libs/libarchfpga/src/physical_types.h

```diff
@@ -251,6 +322,14 @@ struct t_grid_loc_def {
 
     t_grid_loc_spec x;      //Horizontal location specification
     t_grid_loc_spec y;      //Veritcal location specification
+
+    // When 1 metadata tag is split amoung multiple t_grid_loc_def, one
+    // t_grid_loc_def is arbitrarily choosen to own the metadata, and the other
+    // t_grid_loc_def point to the owned version.
```

Comment:
Might want to mention this is a flyweight pattern to help the reader grasp it (your call; I don't feel strongly about it).